### PR TITLE
DB-7482 support multi-probe scan for inlist with expressions

### PR DESCRIPTION
**Pull request by yxia92**
Please see solution description in [DB-7482](https://splicemachine.atlassian.net/browse/DB-7482)

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/InListMultiprobeIT.java [hunk 1]
```diff
@@ -931,7 +994,27 @@ public void testInListWithExpressionsOnPK() throws Exception {
         rs.close();
         ps.close();
 
-        // Q4: prepare statement - inlist condition + other conditions
+        // Q4-1: prepare statement - inlist condition + other conditions
+        /* disable this test for now due to intermittent wrong result tracked in DB-7778 */
```
[the remark this answers is not in the document]
Comment (yxia92) [reply splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/InListMultiprobeIT.java]: Thanks Daniel. It is better to move the commented out test to a separate test. I'll make the change.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryRelationalOperatorNode.java [hunk 2]
```diff
@@ -1423,7 +1423,18 @@ public double getReferenceSelectivity(Optimizable optTable) throws StandardExcep
...
+            // it is possible that this is a special case that actually represents an inlist condition, then it is better to facter in the
+            // number of inlist elements in the selectivity estimation
+            int factor = 1;
+            if (inListProbeSource != null) {
+                factor = inListProbeSource.getRightOperandList().size();
+            }
+            double sel = factor * ((ColumnReference) leftOperand).columnReferenceEqualityPredicateSelectivity();
+            // avoid the estimation to go over 1, in case it happens, round down to 0.9 to be consistent with the logic in
+            // InListSelectivity.getSelectivity()
+            if (sel > 1.0d)
```
[the remark this answers is not in the document]
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryRelationalOperatorNode.java]: Daniel, changing it to 0.9d makes sense. The reason I used 1.0d is to be consistent with the existing logic in InListSelectivity. I'll make change for both places to compare with 0.9d in my next ticket DB-7684 for cost estimation. Thanks.

**Pull request merged**